protocol/statetracker/provider_state_tracker.go

```diff
@@ -104,7 +124,20 @@ func (pst *ProviderStateTracker) LatestBlock() int64 {
 }
 
 func (pst *ProviderStateTracker) GetMaxCuForUser(ctx context.Context, consumerAddress, chainID string, epoch uint64) (maxCu uint64, err error) {
-	return pst.stateQuery.GetMaxCuForUser(ctx, consumerAddress, chainID, epoch)
+	latestBlockTime := pst.EventTracker.getLatestBlockTime()
```

Comment:
state tracker shouldnt modify the responses, it's an interface to fetch the data

Reply:
also event tracker is not the place to save latest block time, there is chain tracker for lava for it